relbench/data/task_base.py

```diff
@@ -183,38 +157,23 @@ class TaskType(Enum):
     """
     REGRESSION = "regression"
     BINARY_CLASSIFICATION = "binary_classification"
+    LINK_PREDICTION = "link_prediction"
 
 
-class RelBenchTask(Task):
```

Comment:
I think we can drop `RelBenchBaseTask` since all it does look like loading prepared cached table dict.
I think we can create the utility functions and call them inside `RelBenchLinkTask` or `RelBenchNodeTask`.